fs/inode/fs_files.c

```diff
@@ -361,7 +361,9 @@ int fs_getfilep(int fd, FAR struct file **filep)
       return -EAGAIN;
     }
 
-  if ((unsigned int)fd >= CONFIG_NFILE_DESCRIPTORS_PER_BLOCK * list->fl_rows)
+  if (fd < 0 || fd >= list->fl_rows * CONFIG_NFILE_DESCRIPTORS_PER_BLOCK ||
```

Comment:
I suggest to `ret = _files_semtake(list);` before accessing `fl_files[][].f_inode`